File: api/ti84_.cxx
/**
 *  Java Grinder
 *  Author: Michael Kohn
 *   Email: mike@mikekohn.net
 *     Web: http://www.mikekohn.net/
 * License: GPL
 *
 * Copyright 2014-2015 by Michael Kohn
 *
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdint.h>

#include "JavaClass.h"
#include "cpu.h"

// FIXME - strncmp?  why?

#define CHECK_FUNC(funct) \
  if (strncmp(#funct, function, sizeof(#funct)-1) == 0) \
  { \
    return generator->ti84_##funct(); \
  }

int ti84(JavaClass *java_class, Generator *generator, char *function)
{
  CHECK_FUNC(clearScreen)
  CHECK_FUNC(clearRect)
  CHECK_FUNC(drawLine)
  CHECK_FUNC(drawPoint)
  CHECK_FUNC(print)
  CHECK_FUNC(printCenter)
  CHECK_FUNC(printHL)
  CHECK_FUNC(putc)
  CHECK_FUNC(fillRect)
  CHECK_FUNC(setCursorX)
  CHECK_FUNC(setCursorY)
  CHECK_FUNC(setDrawBGColor)
  CHECK_FUNC(setDrawBGWhite)
  CHECK_FUNC(setDrawColor)
  CHECK_FUNC(setFillColor)
  CHECK_FUNC(setTextBGColor)
  CHECK_FUNC(setTextColor)

  return -1;
}

File: java/net/mikekohn/java_grinder/TI84.java
/**
 *  Java Grinder
 *  Author: Michael Kohn
 *   Email: mike@mikekohn.net
 *     Web: http://www.naken.cc/
 * License: GPL
 *
 * Copyright 2014-2015 by Michael Kohn
 *
 */

package net.mikekohn.java_grinder;

// http://wikiti.brandonw.net/index.php?title=84PCSE:OS:Applications

abstract public class TI84
{
  protected TI84()
  {
  }

  public static void clearScreen() { }
  public static void clearRect(int x0, int x1, int y0, int y1) { }
  public static void drawLine(int x0, int y0, int x1, int y1, int operation) { }
  public static void drawPoint(int x, int y, int operation) { }
  public static void printHL() { }
  public static void print(String text) { }
  public static void printCenter(String text) { }
  public static void putc(char c) { }
  public static void fillRect(int x0, int x1, int y0, int y1) { }
  public static void setCursorX(int x) { }
  public static void setCursorY(int y) { }
  public static void setDrawBGColor(int color) { }
  public static void setDrawBGWhite() { }
  public static void setDrawColor(int color) { }
  public static void setFillColor(int color) { }
  public static void setTextBGColor(int color) { }
  public static void setTextColor(int color) { }

  public static final int OPERATION_ERASE_PIXEL = 0;
  public static final int OPERATION_DRAW_PIXEL = 1;
  public static final int OPERATION_INVERT_PIXEL = 2;
  public static final int OPERATION_TEST_PIXEL = 3;
  public static final int OPERATION_RESTORE_PIXEL = 4;
}

